gtfs_funnel/schedule_stats_by_route_direction.py

```diff
@@ -12,77 +12,68 @@
 from shared_utils.rt_utils import METERS_PER_MILE
 from update_vars import GTFS_DATA_DICT, SCHED_GCS, RT_SCHED_GCS
 
-def find_most_common_dir(
-    stop_times_gdf: gpd.GeoDataFrame,
-    trips_to_route_df: pd.DataFrame,
-) -> pd.DataFrame:
+def cardinal_direction_for_route_direction(analysis_date:str, dict_inputs:dict):
     """
-    Instead of only having direction_id as 0/1, add back the
-    cardinal directions. 
+    Get a cardinal direction (North, South, East, West) on the route grain.
     """
-    stop_times_col = [
-        "feed_key",
-        "stop_id",
-        "stop_sequence",
-        "schedule_gtfs_dataset_key",
-        "trip_instance_key",
-        "shape_array_key",
-        "stop_name",
-        "prior_stop_sequence",
-        "subseq_stop_sequence",
-        "stop_pair",
-        "stop_pair_name",
-        "stop_primary_direction",
-        "stop_meters",
-    ]
+    STOP_TIMES_FILE = dict_inputs.rt_vs_schedule_tables.stop_times_direction
     
-    # Keep only a subset
-    stop_times_gdf2 = stop_times_gdf[stop_times_col]
+    stop_times_gdf = pd.read_parquet(
+    f"{RT_SCHED_GCS}{STOP_TIMES_FILE}_{analysis_date}.parquet",
+    filters=[[("stop_primary_direction", "!=", "Unknown")]
+    ])
+    
+    trip_scheduled_col = [
+    "route_id",
+    "trip_instance_key",
+    "gtfs_dataset_key",
+    "shape_array_key",
+    "direction_id",
+    "route_long_name",
+    "route_short_name",
+    "route_desc",
+    "name"
+    ]
+        
+    trips_df = helpers.import_scheduled_trips(analysis_date, 
+                                             columns = trip_scheduled_col,
+                                             get_pandas = True)
 
+    
     # Merge dfs
     merge_cols = ["trip_instance_key", 
                   "schedule_gtfs_dataset_key", 
                   "shape_array_key"]
-
-    df1 = pd.merge(stop_times_gdf2, trips_to_route_df, on=merge_cols, how="inner")
+    
+    stop_times_with_trip = pd.merge(stop_times_gdf, trips_df, on = merge_cols)
     
     # Fill in missing direction id with 0, per our usual practice.
-    df1.direction_id = df1.direction_id.fillna(0)
+    stop_times_with_trip.direction_id = stop_times_with_trip.direction_id.fillna(0)
```

Comment:
Do you know if this is going to affect `gtfs_digest/merge_data.py`? Does everything merge or do rows drop off? If rows drop off, this `fillna()` should probably be moved into `merge_data`

Reply:
About 745 rows from the `cardinal_direction_for_route_direction()` drop off when it is merged with `schedule_metrics_by_route_direction()`. I experimented with commenting out `fillna()` and 615 rows still drop off from `cardinal_direction_for_route_direction()`.

Reply:
However, every row from `schedule_metrics_by_route_direction()` merges. There are no `left_only` rows.